data/core/commands/core.lua

```diff
@@ -127,20 +105,20 @@ command.add(nil, {
       local dirname, filename = view.doc.abs_filename:match("(.*)[/\\](.+)$")
       if dirname then
         dirname = core.normalize_to_project_dir(dirname)
-        text = dirname == core.project_dir and "" or common.home_encode(dirname) .. PATHSEP
+        text = dirname == core.root_project().path and "" or common.home_encode(dirname) .. PATHSEP
       end
     end
     core.command_view:enter("Open File", {
       text = text,
       submit = function(text)
-        local filename = system.absolute_path(common.home_expand(text))
+        local filename = core.project_absolute_path(common.home_expand(text))
         core.root_view:open_doc(core.open_doc(filename))
       end,
       suggest = function (text)
-          return common.home_encode_list(common.path_suggest(common.home_expand(text)))
-        end,
+        return common.home_encode_list(common.path_suggest(common.home_expand(text), core.root_project() and core.root_project().path))
```

Comment:
Now that `common.path_suggest` doesn't have a fallback for the second parameter, what should we do here?

Same for the calls in `treeview`.

Reply:
In all cases, we should have a `root_project`, at present. I guess this could change in future, but for now, I think this is fine.